pdns/dnsname.cc

```diff
@@ -113,72 +113,138 @@ DNSName::DNSName(const char* pos, size_t len, size_t offset, bool uncompress, ui
   packetParser(pos, len, offset, uncompress, qtype, qclass, consumed, 0, minOffset);
 }
 
+static void checkLabelLength(uint8_t length)
+{
+  if (length == 0) {
+    throw std::range_error("no such thing as an empty label to append");
+  }
+  if (length > 63) {
+    throw std::range_error("label too long to append");
+  }
+}
+
+// this parses a DNS name until a compression pointer is found
+size_t DNSName::parsePacketUncompressed(const UnsignedCharView& view, size_t pos, bool uncompress)
+{
+  const size_t initialPos = pos;
+  size_t totalLength = 0;
+  unsigned char labellen = 0;
+
+  do {
+    labellen = view.at(pos);
+    ++pos;
+
+    if (labellen == 0) {
+      --pos;
+      break;
+    }
+
+    if (labellen >= 0xc0) {
+      if (!uncompress) {
+        throw std::range_error("Found compressed label, instructed not to follow");
+      }
+      --pos;
+      break;
+    }
+
+    if ((labellen & 0xc0) != 0) {
+      throw std::range_error("Found an invalid label length in qname (only one of the first two bits is set)");
```

Comment:
Can this condition happen? We already know `labellen < 0xc0`.

Reply:
Yes, this condition is true is at least one of the first two bits is set, and `labellen >= 0xc0` when both are set.
`test_compression_single_bit_set` in ` test-dnsname_cc.cc` actually check this, which shows in the code coverage report:
- master: https://coveralls.io/builds/65265194/source?filename=pdns%2Fdnsname.cc#L154
- this PR: https://coveralls.io/builds/65239489/source?filename=pdns%2Fdnsname.cc#L151

Reply:
Right, thinko on my side.